src/Network/WebSockets/TSSocketManager.m

```diff
@@ -127,8 +122,15 @@ - (void)becomeActive
 }
 
 + (void)resignActivity {
-    SRWebSocket *socket = [[self sharedManager] websocket];
-    [socket close];
+    [[self sharedManager] closeWebSocket];
```

Comment:
This is the core fix - `resignActivity` was closing the websocket without nil'ing out the property, nil'ing out its delegate or updating the status property.